sql/rpl_gtid.cc

```diff
@@ -3417,21 +3418,27 @@ my_bool Window_gtid_event_filter::exclude(rpl_gtid *gtid)
 
     if (!m_has_start && is_gtid_at_or_before(&m_stop, gtid))
     {
+      DBUG_ASSERT(m_has_stop);
```

Comment:
This feels a bit late to assert **after** checking `is_gtid_at_or_before(&m_stop, gtid)`. Perhaps to instead make the check
```
    if (!m_has_start && m_stop && is_gtid_at_or_before(&m_stop, gtid))
```